### ISSUE TITLE
Clarify how `:set` affects local values of global-local options

### ISSUE THREAD

**Issue by ddickstein** [vim/vim#12603]
### Steps to reproduce

It's unclear to me whether `:set`ting a global-local value sets both the local and global values or unsets the local value so the global value is used. Different tests indicate different things:

```viml
" Initialize the test.
setg autoread
setl noautoread

" Test what `:set` sets.
set autoread
setg noautoread
set autoread? " Value is `autoread`, indicating that the local value was set.
```

```viml
" Initialize the test
setg tagcase=followic
setl tagcase=ignore

" Test what `:set` sets.
set tagcase=ignore
setg tagcase=followic
set tagcase? " Value is `followic`, indicating `:set` unsets the local value.
```

### Expected behaviour

Consistent behavior and/or clarification of semantics in docs. Is this a difference between `number`/`boolean` and `string` options?

### Version of Vim

9.0.1664

**Comment by chrisbra** [comment]
Yes, case 1 as you expect. For case 2, I think this is again because you choose a *global-local* setting, and the value is unset to indicate that the global value will be use.  If you check with a true local option again, like `tagfunc`, it will behave again like case 1):

```
" Initialize
:setg tagfunc=setg |setl tagfunc=setl 
" Test
set tagfunc=set | setg tagfunc=setg
:echo  &tagfunc
" value for the local option is used
```

**Comment by chrisbra** [comment]
are we fine to close?

**Comment by matveyt** [comment]
@ddickstein is right. There's inconsistency between string and number/boolean options here.

That is, `set makeprg=xxx` is always equivalent to `setg makeprg=xxx` followed by `setl makeprg=` (which is kind of expected). But `set undolevels=100` results in `setg undolevels=100` followed by **either** `setl undolevels=-123456` **or** `setl undolevels=100` depending on whether local undolevels was set earlier or not.

To me it rather looks like a bug.

**Comment by ddickstein** [comment]
Right - I am specifically making a point about inconsistent behaviors between across global-local options. `tagfunc`, which is a buffer-local option, isn't applicable.
